compiler/Makefile

```diff
@@ -23,6 +23,19 @@ ifndef CHPL_MAKE_HOME
 export CHPL_MAKE_HOME=$(shell pwd)/..
 endif
 
+# Generate tags only if $TAGS turned on explicitly, or if $CHPL_DEVELOPER is on
```

Comment:
For the compiler and runtime there's already some code to handle OPTIMIZE and DEBUG in [compiler/make/Makefile.compiler.head] (https://github.com/ben-albrecht/chapel/blob/tags-devonly/compiler/make/Makefile.compiler.head#L19-L28)  and [runtime/make/Makefile.runtime.head] (https://github.com/ben-albrecht/chapel/blob/tags-devonly/runtime/make/Makefile.runtime.head#L19-L27)

I'd guess that's where the code that sets the default of TAGS should be (and that it should use `1` instead of `true` to determine if tags should be generated to match what we do for OPTIMIZE and DEBUG